src/QMCDrivers/WalkerLogManager.cpp

```diff
@@ -68,28 +68,25 @@ WalkerLogManager::WalkerLogManager(WalkerLogInput& inp, bool allow_logs, std::st
 }
 
 
-WalkerLogCollector* WalkerLogManager::makeCollector()
+std::unique_ptr<WalkerLogCollector> WalkerLogManager::makeCollector() const
 {
   if (state.verbose)
     app_log() << "WalkerLogManager::makeCollector " << std::endl;
-  WalkerLogCollector* tc = new WalkerLogCollector(state);
-  return tc;
+  return std::make_unique<WalkerLogCollector>(state);
 }
 
 
-void WalkerLogManager::startRun(std::vector<WalkerLogCollector*>& collectors)
+void WalkerLogManager::startRun(RefVector<WalkerLogCollector>&& collectors)
 {
+  collectors_in_run_ = std::move(collectors);
   if (!state.logs_active)
     return; // no-op for driver if logs are inactive
   if (state.verbose)
```

Comment:
A check (+throw) should be made here for non-empty `collectors` vector.  It is a contradiction for `state.logs_active` and `collectors.size()==0` to be true simultaneously.

Reply:
added